src/api/generics.lisp

```diff
@@ -0,0 +1,65 @@
+(in-package :cl-data-structures)
+
+
+(defgeneric at (container location)
```

Comment:
IMHO at should always return either one or two values to be consistent. Third option is to provide some keyword argument like :default which will be returned when nothing is found. We can discuss preference later.